sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/service/json/JsonConverter.java

```diff
@@ -0,0 +1,375 @@
+package io.stargate.sgv2.docsapi.api.service.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+import com.fasterxml.jackson.databind.node.DoubleNode;
+import com.fasterxml.jackson.databind.node.LongNode;
+import com.fasterxml.jackson.databind.node.NullNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import io.stargate.proto.QueryOuterClass;
+import io.stargate.sgv2.docsapi.api.common.properties.document.DocumentProperties;
+import io.stargate.sgv2.docsapi.config.constants.Constants;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+
+public class JsonConverter {
+  private final ObjectMapper mapper;
+  private final DocumentProperties docsProperties;
+
+  @Inject
+  public JsonConverter(ObjectMapper mapper, DocumentProperties docsProperties) {
+    if (mapper == null) {
+      throw new IllegalStateException("JsonConverter requires a non-null ObjectMapper");
+    }
+    this.mapper = mapper;
+    this.docsProperties = docsProperties;
+  }
+
+  public JsonNode convertToJsonDoc(
+      List<QueryOuterClass.Row> rows,
```

Comment:
Now, instead of providing a list of datastore rows, you must provide a list of grpc rows, along with a list of columns that can be used to determine positional access in the rows.

Reply:
In #1796 I introduced an `ExtendedRow` (name subject to change) wrapper that holds the column definitions, maybe it could be useful here too.